docs_new/functions.md

```diff
@@ -938,6 +938,15 @@ Arguments:
   | ---------- | :------: | ------ | ----------- |
   | instanceID | Yes      | string | ID of RDS instance you want to create snapshot of |
   | dbEngine   | No       | string | Required in case of RDS Aurora instance. Supported DB Engines: `aurora` `aurora-mysql` and `aurora-postgresql` |
+  | credentialsSource | No | string | Source for aws credentials. Supported sources: `profile`, `secret`, `serviceaccount`. Default value is `profile` |
+  | credentialsSecret | No | string | Secret to get credentials from. Only used with `credentialsSource: secret`. Secret with this name should be referenced in the Actionset |
+  | region | No | string | AWS region to use. Derived from profile or serviceaccount if not set |
+
+::: tip NOTE
+
+`credentialsSource: serviceaccount` uses kanister operator service account. IAM role for service
```

Comment:
would it not be a good idea to have another field (`credentialsSA`) like `credentialsSecret`, that can can be used to configured the service account?

Reply:
Maybe. The idea here was that an AWS IAM admin would allow kanister service account access to RDS by creating a role for it.
Associating a different service account would be better because it's not trivial to assign multiple roles to a single SA, but implementation-wise it's harder because we currently run RDS operation from within kanister controller (in the function code) meaning that we can't change SA on the flight there.

We can also try using AssumeRole and configure which role it will assume. This is going to be more flexible with less code changes.

I'd suggest we keep it for the follow-up PRs.

Reply:
ok 👍 .